src/arithmetic/string_funcs/string_funcs.c

```diff
@@ -20,12 +20,14 @@
 
 // returns a string containing the specified number of leftmost characters of the original string.
 SIValue AR_LEFT(SIValue *argv, int argc, void *private_data) {
-	if(SIValue_IsNull(argv[0]) || SIValue_IsNull(argv[1])) return SI_NullVal();
-
-	int64_t newlen = argv[1].longval;
-
+	if(SIValue_IsNull(argv[0])) return SI_NullVal();
+	
+	int64_t newlen = -1;
+	if(SI_TYPE(argv[1]) == T_INT64) {
```

Comment:
why not removing the null option for the second argument from the registration and then this if is not needed?

Reply:
Because if null option is removed for second argument, the output of ```right(null, null)``` will be an error of type mismatch, but the expected output is ```NULL```, without error.